doc/api/packages.md

```diff
@@ -289,29 +289,46 @@ import submodule from 'es-module-package/private-module.js';
 // Throws ERR_PACKAGE_PATH_NOT_EXPORTED
 ```
 
-Entire folders can also be mapped with package exports:
+### Subpath export patterns
+
+> Stability: 1 - Experimental
+
+Explicitly listing each exports subpath entry is recommended for packages with
+a small number of exports. But for packages that have very large numbers of
+subpaths this can start to cause package.json bloat and maintenanec issues.
+
+For these use cases, subpath export patterns can be used instead:
 
 ```json
 // ./node_modules/es-module-package/package.json
 {
   "exports": {
-    "./features/": "./src/features/"
+    "./features/*": "./src/features/*.js"
```

Comment:
Confirming: `*` matches `/` as well? Would maybe we want to make `*` not match `/`, and `**` match `/`? That way you could control whether you had nested patterns or not.

Reply:
I think I would rather make it `(.*)` since that's the other common (and more modern) way to write "splats" in route definitions. The current syntax matches what express <=4 (?) used. I think using glob-style patterns would imply expansion, not 1-to-1 mapping.

Reply:
These are file paths, not route definitions, in which double-star globbing is exceedingly common. Why would express routing patterns be appropriate here?

Reply:
Exports patterns are not file paths, they are effectively a relative URL space (including URI encoding) that apply a mapping to a file space, so routes are a fairly good analogy.

Reply:
If it's going to match express' routing DSL, then that's a different story (including path segment awareness), but I don't think it would make sense to only partially mimic express.

Reply:
@ljharb `*` will match any separators yes. But if anything the right hand side is more what should be thought of as a glob since replacing the `*` in the right hand side with `**` is what will give you the file system glob to enumerate the exports of the package.

Your comment also triggered another question though which I hadn't considered. And that is that:

```json
{
  "exports": {
    "./*": "./*"
  }
}
```

is clearly defined to only match explicit subpaths of the package. That is, `import.meta.resolve('pkg/')` would not resolve that export mapping (currently it throws with this PR, unlike "./") and therefore, interestingly, if `./` path mappings are deprecated could this could be extended to always resolve the package base in a subsequent PR if we want to.

Reply:
I'm confused, why would that be different? A `*` can always be empty.

Reply:
This PR explicitly requires some text to be matched on the LHS for the `*`. If you write:

```json
{
  "exports": {
    "./components-*": "./components/*.js"
  }
}
```

it will match `pkg/components-x` but `pkg/components-` will throw a package path not exported error instead of trying to map into `components/.js` and throwing a not found.

Reply:
That seems very confusing, and not how `*` works in literally every other context everywhere.